### DISCUSSION TITLE
Idea for tutorial

### DISCUSSION THREAD

**Discussion by portfedh** [py-pdf/fpdf2#569] [Ideas]
Hi!

I just made a short blog post of a snippet i use a lot with fpdf2. 

Its a script to merge odd and even pdf pages. 

I use it when I scan files with lots of pages on a scanner that can only scan one side at a time. 

I think its simple enough that it might be a good addition to the fpdf2 tutorial section. 

Here is the blog post. 
https://pablocruz.io/merging-odd-and-even-pages-in-a-pdf/

Let me know what you think!
Pablo

**Comment by gmischler** [comment]
> `from PyPDF2 import PdfFileReader, PdfFileWriter`

While your blog post instructs to install fpdf2, what your code actually uses is an entirely different package...

**Comment by portfedh** [comment]
Thank you for pointing that out!

I made the same script with several packages and uploaded the wrong one. 
I'll correct the post and post again for you to review.

**Comment by portfedh** [comment]
I double checked, and found the reason I ended up switching in to PyPDF2. 

In FPDF2, I found this tutorial.
https://pyfpdf.github.io/fpdf2/ExistingPDFs.html

I tried to work on it, but even after installing, i got:
>>> ModuleNotFoundError: No module named 'pdfrw'

Sorry about that. I still think its a good idea, but I couldn't get it to work. 
I'll change the post to reflect the right package.  : (
I

**Reply by gmischler** [reply]
> In FPDF2, I found this tutorial. https://pyfpdf.github.io/fpdf2/ExistingPDFs.html

At the very top of that page, it explains:

>>This page provides several examples of doing so **using [pdfrw](https://github.com/pmaupin/pdfrw)**, ...

So why didn't you just install the pdfrw package at that point?

Of course, if you only want to shuffle pages in existing PDFs around, then using a full fledged PDF **generator** like fpdf2 is overkill in the first place.
Each one of pdfrw *or* PyPDF2 will do that just fine on its own.

**Comment by portfedh** [comment]
I did try to install using: 
    pip install pdfrw

And also the way the tutorial mentions in fpdf2:
    https://pyfpdf.github.io/fpdf2/ExistingPDFs.html
    pip install git+https://github.com/PyFPDF/pdfrw.git@addpage_at_index

But when running the script I kept getting an error as if the library was not installed. I didn't really get why.

I agree, this is a very simple thing, and its overkill.
